DEVELOPMENT.md

```diff
@@ -1,119 +1,197 @@
-# Development
+# Development Guide
+
+This guide will help you set up the development environment, deploy the backend, and configure necessary integrations and credentials.
+
+<!-- TOC -->
+* [Development Guide](#development-guide)
+  * [Prerequisites](#prerequisites)
+  * [Setup](#setup)
+    * [1. Install Firebase CLI](#1-install-firebase-cli)
+    * [2. Install Project Dependencies](#2-install-project-dependencies)
+    * [3. Run the Project Locally](#3-run-the-project-locally)
+  * [Deployment](#deployment)
+    * [1. Login to Firebase](#1-login-to-firebase)
+    * [2. Deploy Project to Firebase](#2-deploy-project-to-firebase)
+  * [Optional Local Setup for Firebase Admin SDK](#optional-local-setup-for-firebase-admin-sdk)
+    * [Setting Up Credentials](#setting-up-credentials)
+      * [Linux / MacOS](#linux--macos)
+      * [Windows (PowerShell)](#windows-powershell)
+  * [Integrations & Environment Variables](#integrations--environment-variables)
+    * [Discord Integration](#discord-integration)
+    * [GitHub Integration](#github-integration)
+    * [Internal Token (for Self-Authentication)](#internal-token-for-self-authentication)
+  * [Updating Configuration Variables in Firebase](#updating-configuration-variables-in-firebase)
+  * [Local Development Commands](#local-development-commands)
+  * [Troubleshooting & Tips](#troubleshooting--tips)
+  * [Contributing](#contributing)
+<!-- TOC -->
+
+## Prerequisites
+
+Ensure you have the following installed:
+
+- **Node.js** (Recommended: Version 20 or later)
+- **Yarn** (Package manager)
 
 ## Setup
 
-Install firebase globally
+### 1. Install Firebase CLI
+
+To interact with Firebase, install the Firebase CLI globally:
 
 ```bash
-npm i -g firebase-tools
+yarn global add firebase-tools
```

Comment:
In this current case, I think both commands will achieve the same outcome. `yarn global add` is pretty much the same as `npm i -g`. Since we're using yarn everywhere, I think it's ok to also use it to install global things.

Maybe we should actually add a step to install yarn globally first tho. :+1:

Reply:
Or do you have specific cases where `yarn global add` won't work on some systems?

Reply:
I've added a link to yarn installation in 32abcc5 so we don't have to explain how to install it.

Reply:
Yea 

Firstly: there's cases where `yarn global add` either doesn't work or where the installed package isn't globally accessible out of the box, the same way `npm` does.

- For example, there's some edge cases for different Linux OSes where yarn can be installed using the package manager vs with npm vs yet other ways, where packages may behave differently or not be recognised globally. 
- In Windows I've personally had to revert back to `npm` for global packages because of several issues - the most common one being that the installed packages weren't automatically added to the global path (i.e. not accessible by default)

Secondly: Yarn has a large delta in functionality between each major version. The command you used works in yarn 1, but was removed in 2.x. 

Without going into too much nitty gritty: Experience has shown that Yarn is excellent for a project level package manager. Yarn is quick to type in, plays well with other tools like `Volta` and `corepack`, and it has been progressively maintained over the years. For system level nothing beats the good old `npm i -g` as it works on any platform regardless of how it was installed.

Reply:
```suggestion
npm i -g firebase-tools
```